### PR TITLE
Code to add license-header flag to autorest v6

### PR DESCRIPTION
**Pull request by sarangan12**
@joheredi I wanted to try a simple feature to start the code gen development. So, I picked up issue https://github.com/Azure/autorest.typescript/issues/555 

Added code change, test swagger and test code for the same. Please take a look and provide your feedback.

@ramya-rao-a FYI

### CODE REVIEW

**Comment by joheredi** [comment]
Looks good! Can you run

'npm run test' followed by

'Npm run checkout-specs && npm run smoke-test'

And add the changed files to your PR please?

File: tsconfig.browser-test.json [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
+  },
+  "include": [
+    "test/integration/*.ts",
+    "test/integration/generated/**/*.ts",
+    "test/utils/stream-helpers.ts"
+  ],
+  "exclude": [
+    "test/integration/generated/noLicenseHeader/*.ts",
+    "test/integration/generated/licenseHeader/*.ts",
+    "test/integration/licenseHeader.spec.ts"
```
Comment (chradek) [inline tsconfig.browser-test.json:33]: Just curious, why did you exclude these?

**Comment by sarangan12** [comment]
The recent commits have the following changes: 

1. Linked the license-header flag with the generation of LICENSE.txt file. 
2. Added test cases to check for the presence and absence of the LICENSE.txt file. 
3. Ran `npm run clone:specs && npm run smoke-test` coomands. (No changes)

File: test/utils/test-swagger-gen.ts [hunk 2]
```diff
@@ -1,305 +1,359 @@
...
-  const isDebugging = process.argv.indexOf("--debug") !== -1;
-  buildWhitelist();
-  await buildAutorest();
-  await generateSwaggers(whiteList, isDebugging);
-};
-
-run().catch((error) => {
-  process.exit(-1000);
-});
+import { exec } from "child_process";
```
Comment (joheredi) [inline test/utils/test-swagger-gen.ts:1]: Are there any changes in this file? Or just formatting?

File: src/typescriptGenerator.ts [hunk 3]
```diff
@@ -67,10 +67,13 @@ export async function generateTypeScriptLibrary(
...
   };
 
+  const shouldGenerateLicenseHeader: boolean =
+    (await host.GetValue("license-header")) || false;
+
   // Skip metadata generation if `generate-metadata` is explicitly false
   if ((await host.GetValue("generate-metadata")) !== false) {
     generatePackageJson(clientDetails, packageDetails, project);
-    generateLicenseFile(project);
+    if (shouldGenerateLicenseHeader) generateLicenseFile(project);
```
Comment (joheredi) [inline src/typescriptGenerator.ts:76]: nit: you could pass `shouldGenerateLicenseHeader` to `generateLicenseFile` and check it there, just to maintain the format of just calling the generate functions

File: test/utils/test-swagger-gen.ts (hunk 2 above)
Comment (sarangan12) [reply test/utils/test-swagger-gen.ts:1]: Just formatting

File: src/typescriptGenerator.ts (hunk 3 above)
Comment (sarangan12) [reply src/typescriptGenerator.ts]: Done

**Pull request merged**